edgedb/datatypes/namedtuple.c

```diff
@@ -58,62 +60,34 @@ EdgeNamedTuple_New(PyObject *desc)
         return NULL;
     }
 
-    EdgeNamedTupleObject *nt = NULL;
-    EDGE_NEW_WITH_FREELIST(EDGE_NAMED_TUPLE, EdgeNamedTupleObject,
-                           &EdgeNamedTuple_Type, nt, size);
+    PyTupleObject *nt = NULL;
+    EDGE_NEW_WITH_FREELIST(EDGE_NAMED_TUPLE, PyTupleObject, type, nt, size);
     assert(nt != NULL);
-    assert(EdgeNamedTuple_Check(nt));
+    if (Py_TYPE(nt) != type) {
+        Py_DECREF(Py_TYPE(nt));
+        Py_INCREF(type);
+        Py_TYPE(nt) = type;
+    }
```

Comment:
When we are switching in and out of the freelist, make sure the `ob_type` is still what we expected.

Reply:
This is now done in a different way - we always DECREF the type before putting memory into the freelist, and set and INCREF the type when resuing memory.